verification/src/http_server/handlers/verification/sourcify/api.rs

```diff
@@ -0,0 +1,97 @@
+use crate::{VerificationResponse, VerificationResult};
+use actix_web::{error, error::Error};
+
+use super::types::{ApiFilesResponse, ApiRequest, ApiVerificationResponse, Files};
+
+#[async_trait::async_trait]
+pub(super) trait SourcifyApi {
+    async fn verification(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiVerificationResponse, reqwest::Error>;
+
+    async fn source_files(&self, params: &ApiRequest) -> Result<ApiFilesResponse, reqwest::Error>;
+}
+
+pub(super) struct SoucifyApiClient {
+    host: String,
+}
+
+impl SoucifyApiClient {
+    pub fn new(host: &str) -> Self {
+        Self {
+            host: host.to_string(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl SourcifyApi for SoucifyApiClient {
+    async fn verification(
+        &self,
+        params: &ApiRequest,
+    ) -> Result<ApiVerificationResponse, reqwest::Error> {
+        let resp = reqwest::Client::new()
+            .post(&self.host)
+            .json(&params)
+            .send()
+            .await?;
+
+        resp.json().await
+    }
+
+    async fn source_files(&self, params: &ApiRequest) -> Result<ApiFilesResponse, reqwest::Error> {
+        let url = format!(
+            "{host}/files/any/{chain}/{address}",
+            host = self.host,
+            chain = params.chain,
+            address = params.address,
+        );
+        let resp = reqwest::get(url).await?;
+
+        resp.json().await
+    }
+}
+
+pub(super) async fn verify_using_sourcify_client(
+    sourcify_client: impl SourcifyApi,
+    params: ApiRequest,
+) -> Result<VerificationResponse, Error> {
+    let response = sourcify_client
+        .verification(&params)
+        .await
+        .map_err(error::ErrorInternalServerError)?;
+
+    match response {
+        ApiVerificationResponse::Verified { result: api_result } => {
+            let files = {
```

Comment:
I suggest may be moving each match arm into separate function

Reply:
I removed unclear logic from it, so i think there is no need to separate into functions